deploy/kurl/kotsadm/template/base/install.sh

```diff
@@ -499,6 +499,62 @@ function kotsadm_health_check() {
     return 0
 }
 
+# check statefulset readiness
+function check_stateful_set_readiness() {
+    statefulsetName=$1
+    statefulSetPodVersionLabel="controller-revision-hash"
+
+    # get the list of pods for the statefulset
+    pods=$(kubectl get pods -l "app=$statefulsetName" -o jsonpath="{.items[*].metadata.name}")
+
+    # validate if each pod has the same version label, owner reference and is running
+    currentVersion=""
+    for pod in $pods; do
+        ownerAndVersionAndPhase=$(kubectl get pod $pod -o=jsonpath='{.metadata.ownerReferences[?(@.kind=="StatefulSet")].name} {.metadata.labels.'$statefulSetPodVersionLabel'} {.status.phase}')
+
+        # check if the owner reference is the statefulset
+        owner=$(echo $ownerAndVersionAndPhase | awk '{print $1}')
+        if [ "$owner" != "$statefulsetName" ]; then
+            return 1
+        fi
+
+        # get the version label value
+        version=$(echo $ownerAndVersionAndPhase | awk '{print $2}')
+        if [[ -z "$version" ]]; then
+            exit 1
+        fi
+
+        # check if the version label is the same as the current version
+        if [[ -z "$currentVersion" ]]; then
+            currentVersion=$version
+        else
+            if [ "$currentVersion" != "$version" ]; then
+                return 1
+            fi
+        fi
+
+        # check if the pod is running
+        phase=$(echo $ownerAndVersionAndPhase | awk '{print $3}')
+        if [ "$phase" != "Running" ]; then
+            return 1
+        fi
+    done
+
+    # check statefulset readiness
+    replicas=$(kubectl get statefulset $statefulsetName -o=jsonpath='{.spec.replicas} {.status.readyReplicas}')
+    desiredReplicas=$(echo $replicas | awk '{print $1}')
+    if [[ -z "$desiredReplicas" ]]; then
+        desiredReplicas=1
+    fi
+
```

Comment:
This is a very unlikely scenario with a kurl install, but it did come up in the status informers, so I'll mention it here: there can be a moment when there are n-1 pods that are ready and all share the same revision right before a new pod is deployed with a different revision. So in a super rare situation, while the pod revisions are being checked a new pod in the stateful set comes up with the wrong revision and then the ready replicas check passes, but things are in a bad state.

Again, very unlikely to occur in a kurl install, but might be worth considering changing this logic to:
* get current revision
* get replicas
* get desired replicas
* if replicas == desired replicas then get pod list for current revision
* if length of pod list == desired replicas then check that all pods are the correct revision